pkg/graph/resolvers/tag.go

```diff
@@ -0,0 +1,148 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/appcontext"
+	"github.com/cmsgov/mint-app/pkg/authentication"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
+	"github.com/cmsgov/mint-app/pkg/userhelpers"
+)
+
+// TaggedHTMLGet returns the tag content of a parent object
+func TaggedHTMLGet(
+	logger *zap.Logger,
+	store *storage.Store,
+	rawContent string, // if desired, we could get this from the database as well
+	taggedTable string,
+	taggedField string,
+	taggedContentID uuid.UUID) (*models.TaggedHTML, error) {
+	taggedHTML, err := models.NewTaggedHTMLFromString(rawContent)
+	if err != nil {
+		return nil, err
+	}
+
+	tags, err := TagCollectionGet(logger, store, taggedTable, taggedField, taggedContentID)
+	if err != nil {
+		return nil, err
+	}
+	taggedHTML.Tags = tags
+
+	return &taggedHTML, nil
+
+}
+
+// TagCollectionGet retrieves all the tags for a specific entry and table
+func TagCollectionGet(
+	logger *zap.Logger,
+	store *storage.Store,
+	taggedTable string,
+	taggedField string,
+	taggedContentID uuid.UUID) ([]*models.Tag, error) {
+
+	tags, err := store.TagCollectionGetByContentIDAndField(logger, taggedTable, taggedField, taggedContentID)
+
+	return tags, err
+}
+
+// TaggedEntityGet returns a Tagged Entity based on the table it refers to as well as the ID
+func TaggedEntityGet(
+	ctx context.Context,
+	store *storage.Store,
+	tagType models.TagType,
+	EntityUUID *uuid.UUID,
+	EntityIntID *int,
+) (models.TaggedEntity, error) {
+	logger := appcontext.ZLogger(ctx)
+	switch tagType {
+	case models.TagTypeUserAccount:
+		return UserAccountGetByIDLOADER(ctx, *EntityUUID)
+	case models.TagTypePossibleSolution:
+		return PossibleOperationalSolutionGetByID(logger, store, *EntityIntID)
+	default:
+		return nil, fmt.Errorf(" no tagged entity configured for table: %s", tagType)
+	}
+
+}
+
+// UpdateTaggedHTMLMentionsAndRawContent updates the tagged html with the correct entity ids, and updates the RAW HTMl with the new representation of the mentions
+func UpdateTaggedHTMLMentionsAndRawContent(ctx context.Context, store *storage.Store, tHTML *models.TaggedHTMLInput, getAccountInformation userhelpers.GetAccountInfoFunc) error {
```

Comment:
I'd like some thoughts on this function please. I'd almost make it a method, but it requires the store.  If it was made a method, we could perhaps pass another function that abstracts getPossibleOperationalSolutionByKey store method.

The benefit to a method is that the function modifies the `TaggedHTMLInput`. I wonder if that would be cleaner? Or perhaps we could return the modified tagged html instead of modifying the instance?

Reply:
I think this approach is generally fine! I think one of two things would help clarfify expectations of this function:

1. Add comments to the function definition that clarify that the `tHTML` input type _will_ be modified, and that that's the expectation of the function
2. (As you suggested) update the function to return `(tHTML, error)`, and make the code _not_ modify the `*tHTML` input (in this case, I'd suggest updating the comments as well just to be extra clear!)

Reply:
Thanks, I updated the comment to make it even clearer that the tHTML and mention are updated in this function call.

I also extracted some of these functions in your suggstion Tom. I left some code part of the orchestrator where I thought it was more readable to do so. You'll also notice the update methods are combined, as you can't update the individual mention content before updating the parent.